### ISSUE TITLE
Compiler error created by running clippy fix on floating point expression

### ISSUE THREAD

**Issue by MathiasKindberg** [rust-lang/rust#105776]
I got linked here after running `clippy fix` when upgrading our codebase to 1.66.

`cargo clippy --all-features --tests --fix`

I tried this code:

```bash
warning: failed to automatically apply fixes suggested by rustc to crate `algo`

after fixes were automatically applied the compiler reported errors within these files:

  * crates/foo/src/bar.rs

This likely indicates a bug in either rustc or cargo itself,
and we would appreciate a bug report! You're likely to see
a number of compiler warnings after this message which cargo
attempted to fix but failed. If you could open an issue at
https://github.com/rust-lang/rust/issues
quoting the full output of this command we'd be very appreciative!
Note that you may be able to make some more progress in the near-term
fixing code with the `--broken-code` flag

The following errors were reported:
error[E0689]: can't call method `ln` on ambiguous numeric type `{float}`
   --> crates/foo/src/bar.rs:163:51
    |
163 |         let foo= 1.0 * (1.0 / (1.0 / 12.0)).ln();
    |                                                   ^^

error: aborting due to previous error

For more information about this error, try `rustc --explain E0689`.
Original diagnostics will follow.

warning: casting to the same type is unnecessary (`f64` -> `f64`)
   --> crates/algo/src/categorical_relative_entropy.rs:163:30
    |
163 |         let bar= 1.0 * ((1.0 / (1.0 / 12.0)) as f64).ln();
    |                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: try: `(1.0 / (1.0 / 12.0))`
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#unnecessary_cast
    = note: `#[warn(clippy::unnecessary_cast)]` on by default
```

After that is a bunch of fixes when formatting strings.

```bash
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#uninlined_format_args
help: change this to
435 -         println!("foo={:?}", foo);
435 +         println!("foo={foo:?}");
```

And some warnings for elided lifetimes.
```bash
warning: the following explicit lifetimes could be elided: 'a
   --> crates/foo/src/bar/mod.rs:258:5
    |
258 | /     fn baz<'a, 'b>(
259 | |         &'a mut self,
260 | |         batch: &'b Foo,
261 | |     ) -> anyhow::Result<Bar<'b>> {
    | |_________________________________________^
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#needless_lifetimes`
```
